crates/iceberg/src/transaction/mod.rs

```diff
@@ -104,32 +110,13 @@ impl Transaction {
     }
 
     /// Sets table to a new version.
-    pub fn upgrade_table_version(mut self, format_version: FormatVersion) -> Result<Self> {
-        let current_version = self.current_table.metadata().format_version();
-        match current_version.cmp(&format_version) {
-            Ordering::Greater => {
-                return Err(Error::new(
-                    ErrorKind::DataInvalid,
-                    format!(
-                        "Cannot downgrade table version from {} to {}",
-                        current_version, format_version
-                    ),
-                ));
-            }
-            Ordering::Less => {
-                self.apply(vec![UpgradeFormatVersion { format_version }], vec![])?;
-            }
-            Ordering::Equal => {
-                // Do nothing.
-            }
-        }
-        Ok(self)
+    pub fn upgrade_table_version(&self) -> UpgradeFormatVersionAction {
+        UpgradeFormatVersionAction::new()
     }
 
     /// Update table's property.
-    pub fn set_properties(mut self, props: HashMap<String, String>) -> Result<Self> {
```

Comment:
Ooops, it's an interface change, one minor suggestion, better to call it out in PR description and later release notes. :)
It actually breaks [moonlink](https://github.com/Mooncake-Labs/moonlink) compilation :(

Reply:
Yes, we are making `TransactionAction`s retryable and had to change the existing APIs. There are more API changes coming for other actions like `ReplaceSortOrderAction`(https://github.com/apache/iceberg-rust/pull/1441) and `FastAppendAction`(WIP) 

We definitely need to call this out in the next release notes. cc: @liurenjie1024 

I'm not very familiar with moonlink and here goes the dumb question: Would it be better for moonlink to depend on a released/stable iceberg-rs version instead of tracking an unreleased git rev? You can set up a non-blocking github CI to track `iceberg-rs/main` if you want to catch compatibility issues early

Reply:
> We definitely need to call this out in the next release notes.

Thanks!

Reply:
> I'm not very familiar with moonlink and here goes the dumb question: Would it be better for moonlink to depend on a released/stable iceberg-rs version instead of tracking an unreleased git rev?

That's the ideal case, but I think the interval between iceberg-rust releases are too long.